spec/list_all_game_and_author_spec.rb

```diff
@@ -0,0 +1,58 @@
+require_relative '../classes/game'
+require_relative '../classes/author'
+require_relative '../classes/list_all_game_and_author'
+
+RSpec.describe GameAuthorAddAndShow do
+  let(:game_author) { GameAuthorAddAndShow.new }
+
+  describe '#add_game' do
+    it 'adds a game and author to the collection' do
+      game_author = GameAuthorAddAndShow.new
+      allow(game_author).to receive(:gets).and_return(
+        "last_played_at_value\n",
+        "multiplayer_value\n",
+        "2023-06-13\n",
+        "first_name_value\n",
+        "last_name_value\n"
+      )
+
+      expect { game_author.add_game }.to output(/Game added successfully/).to_stdout
+    end
+  end
+
+  describe '#list_all_games' do
+    it 'displays information for all games' do
+      game_author = GameAuthorAddAndShow.new
+
+      allow(game_author).to receive(:load_json).with('data/games.json').and_return([
+                                                                                     { 'id' => 1,
+                                                                                       'last_played_at' => 'last_play',
+                                                                                       'multiplayer' => 'multiplayer_1',
+                                                                                       'publish_date' => '2023-06-13' },
+                                                                                     { 'id' => 2,
+                                                                                       'last_played_at' => 'last_play',
+                                                                                       'multiplayer' => 'multiplayer_2',
+                                                                                       'publish_date' => '2023-06-14' }
+                                                                                   ])
+
+      output = ''
+      allow(game_author).to receive(:display_message) { |message| output << message << "\n" }
```

Comment:
I tested your code, and unfortunately, The `list all games` can't pass the test. when I tried to show list of all games it looks like awkward photo below:
![2](https://github.com/AbdusSattar-70/Ruby-Group-Capstone/assets/118761303/bb11eda5-611a-4ab2-948e-b8dfeb2cd0dd)
![3](https://github.com/AbdusSattar-70/Ruby-Group-Capstone/assets/118761303/32ea92de-f4e4-4c70-99ef-1373f95566f1)